typescript/api/services/UsersService.ts

```diff
@@ -1033,7 +1027,7 @@ export module Services {
      */
     public addUserAuditEvent = (user, action, additionalContext) => {
       let auditEvent = {}
-      if (!_.isEmpty(user.password)) {
+      if (user && !_.isEmpty(user.password)) {
```

Comment:
Occassionally I'd get an error about trying to access `.password` on `undefined`.
This one change won't help the lines below it, there's likely more to do here.
I'm not sure this should be part of this PR.

Reply:
Yes this pops up occasionally in logs as well. user should never be undefined/null here, i suspect its something to do with this function running asynchronously and the reference that user was pointing to has been made null due to a logout (or something similar for the login)

Reply:
I've removed this change from this PR. I think this needs to be addressed separately.